src/ethereum/EthereumVerifier.js

```diff
@@ -353,24 +355,24 @@ class EthereumVerifier {
    */
   static selectWorkerGroup(secretContractAddress, params, workerGroupSize) {
     // Find total number of staked tokens for workers
-    const tokenCpt = params.balances.reduce((a, b) => a + b, 0);
+    const tokenCpt = params.balances.reduce((a, b) => JSBI.add(a, b), JSBI.BigInt(0));
     let nonce = 0;
     const selectedWorkers = [];
     do {
       // Unique hash for epoch, secret contract address, and nonce
-      const hash = cryptography.soliditySha3(
-        {t: 'uint256', v: params.seed},
-        {t: 'bytes32', v: secretContractAddress},
-        {t: 'uint256', v: nonce});
+      const hash = cryptography.hash(abi.rawEncode(
```

Comment:
Can you elaborate on why the change in function?
this adds another dependency

Reply:
Because that's what the [contract](https://github.com/enigmampc/enigma-contract/blob/3048db58f1ccc0fe133e168f0e64e62b7322ce65/contracts/impl/WorkersImpl.sol#L240) and the [library](https://github.com/enigmampc/enigma-contract/blob/3048db58f1ccc0fe133e168f0e64e62b7322ce65/enigma-js/src/Enigma.js#L350) implement, and we need to match such implementations. We are not using `soliditySha3()` because under the hood that used `encodePacked`.